### PR TITLE
Properly rework Necrotic Metabolism

### PR DESCRIPTION
**Pull request by Absolucy**
## About The Pull Request

Partially reverts and reworks https://github.com/BeeStation/BeeStation-Hornet/pull/9669 to how I originally suggested it should work, with a whitelist of symptoms that are okay to tick post-death (none of which are "spreading" symptoms).

## Why It's Good For The Game

Because several symptoms have effects _which are intended to occur post-death_, which can't happen anymore if you just set `process_dead` to always FALSE.

And yes, I know, viro is "supposedly destined for the bin" or whatever, but well, let's have it remain usable to its fullest extent while we still have it.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

2023-12-20 (1703051995) ~ dreamseeker
2023-12-20 (1703052035) ~ dreamseeker
2023-12-20 (1703052082) ~ dreamseeker
2023-12-20 (1703052103) ~ dreamseeker
2023-12-20 (1703052135) ~ dreamseeker

</details>

## Changelog
:cl:
balance: Necrotic Metabolism will now once again allow some symptoms to process in dead hosts, although not all of them.
/:cl:

### CODE REVIEW

File: code/datums/diseases/advance/symptoms/viral.dm [hunk 1]
```diff
@@ -80,10 +82,10 @@ Bonus
 */
 
 /datum/symptom/viralreverse
-
 	name = "Viral Aggressive Metabolism"
 	desc = "The virus sacrifices its long term survivability to nearly instantly fully spread inside a host. \
 	The virus will start at the last stage, but will eventually decay and die off by itself."
+	symptom_flags = SYMPTOM_DEAD_TICK_WEAK
```
Comment (PowerfulBacon) [inline code/datums/diseases/advance/symptoms/viral.dm:88]: The to_chat message should only happen when alive

File: code/datums/diseases/advance/symptoms/skin.dm [hunk 2]
```diff
@@ -351,6 +350,7 @@ Thresholds
 /datum/symptom/spiked
 	name = "Cornu Cutaneum"
 	desc = "The virus causes the host to unpredictably grow and shed sharp spines, damaging those near them."
+	symptom_flags = SYMPTOM_DEAD_TICK_WEAK // the spikes aren't gonna just instantly fall off when you croak
```
Comment (PowerfulBacon) [inline code/datums/diseases/advance/symptoms/skin.dm:353]: The to_chat messages should be disabled on dead people

File: code/datums/diseases/advance/symptoms/radiation.dm [hunk 3]
```diff
@@ -46,6 +46,7 @@
 /datum/symptom/radconversion
 	name = "Aptotic Culling"
 	desc = "The virus causes infected cells to die off when exposed to radiation, causing open wounds to appear on the host's flesh. The end result of this process is the removal of radioactive contamination from the host."
+	symptom_flags = SYMPTOM_DEAD_TICK_NECROTIC
```
Comment (PowerfulBacon) [inline code/datums/diseases/advance/symptoms/radiation.dm:49]: This one is probably better without dead processing considerings its activate functions

**Review by PowerfulBacon — CHANGES_REQUESTED** [review]
All the to_chat stuff needs to not show to dead people, and I kind of think we should avoid letting symptoms that deal damage or modify the body function while dead.

**Comment by Rukofamicom** [comment]
> I know, viro is "supposedly destined for the bin" or whatever, but well, let's have it remain usable to its fullest extent while we still have it.

I disagree with it needing to be available to its fullest extent - necrotic metabolism is better off being removed itself while we wait on the removal.

🔑

File: code/datums/diseases/advance/symptoms/viral.dm (hunk 1 above)
Comment (Rukofamicom) [reply code/datums/diseases/advance/symptoms/viral.dm:88]: Viral Aggressive Metabolism + functioning Necrotic + deadly symptoms is not an interactive way to antagonize rounds at all. This is right up there with plasma flooding in terms of fun for the victims and ruining rounds.

Unless this is continuing to tick only for the sake of destroying itself and not making it actively easier to spread while dead, this one shouldn't be included.

**Pull request closed without merging**